### ISSUE TITLE
panic err using the has-one example code

### ISSUE THREAD

**Issue by frederikhors** [uptrace/bun#520]
Trying the code in https://github.com/uptrace/bun/blob/master/example/rel-has-one/main.go I'm having:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x8a3af7]

goroutine 1 [running]:
modernc.org/sqlite.(*Driver).Open(0x0, {0x9006de?, 0x1000?})
    C:/Users/Fred/go/pkg/mod/modernc.org/sqlite@v1.16.0/sqlite.go:1466 +0x57
database/sql.dsnConnector.Connect(...)
    C:/Program Files/Go/src/database/sql/sql.go:761
database/sql.(*DB).conn(0xc00001eea0, {0x974968, 0xc000012040}, 0x1)
    C:/Program Files/Go/src/database/sql/sql.go:1395 +0x782
database/sql.(*DB).exec(0x600000?, {0x974968, 0xc000012040}, {0xc0001f3000, 0x50}, {0x0, 0x0, 0x0}, 0x1?)
    C:/Program Files/Go/src/database/sql/sql.go:1657 +0x5d
database/sql.(*DB).ExecContext(0x0?, {0x974968, 0xc000012040}, {0xc0001f3000, 0x50}, {0x0, 0x0, 0x0})
    C:/Program Files/Go/src/database/sql/sql.go:1635 +0xe5
github.com/uptrace/bun.(*baseQuery).exec(0xc00019d180, {0x974968?, 0xc000012040?}, {0x974a48?, 0xc00019d180?}, {0xc0001f3000, 0x50})
    C:/Users/Fred/go/pkg/mod/github.com/uptrace/bun@v1.1.4/query_base.go:581 +0xe4
github.com/uptrace/bun.(*CreateTableQuery).Exec(0xc00019d180, {0x974968, 0xc000012040}, {0x10?, 0xb9e600?, 0xc00018c980?})
    C:/Users/Fred/go/pkg/mod/github.com/uptrace/bun@v1.1.4/query_table_create.go:306 +0xad
main.createSchema({0x974968, 0xc000012040}, 0xc000058300)
    D:/go/bun/has-one/main.go:67 +0x12f
main.main()
    D:/go/bun/has-one/main.go:40 +0x2d9
```

Windows 10, Go 1.18.1.

- **main.go**: the same of the example.

- **go.mod**:

<details>
 <summary>Open for the go.mod code</summary>

```go.mod
module has-one

go 1.18

require (
	github.com/uptrace/bun v1.1.4
	github.com/uptrace/bun/dialect/sqlitedialect v1.1.4
	github.com/uptrace/bun/driver/sqliteshim v1.1.4
	github.com/uptrace/bun/extra/bundebug v1.1.4
)

require (
	github.com/fatih/color v1.13.0 // indirect
	github.com/google/uuid v1.3.0 // indirect
	github.com/jinzhu/inflection v1.0.0 // indirect
	github.com/kballard/go-shellquote v0.0.0-20180428030007-95032a82bc51 // indirect
	github.com/mattn/go-colorable v0.1.12 // indirect
	github.com/mattn/go-isatty v0.0.14 // indirect
	github.com/mattn/go-sqlite3 v1.14.12 // indirect
	github.com/remyoudompheng/bigfft v0.0.0-20200410134404-eec4a21b6bb0 // indirect
	github.com/tmthrgd/go-hex v0.0.0-20190904060850-447a3041c3bc // indirect
	github.com/vmihailenco/msgpack/v5 v5.3.5 // indirect
	github.com/vmihailenco/tagparser/v2 v2.0.0 // indirect
	golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3 // indirect
	golang.org/x/sys v0.0.0-20220422013727-9388b58f7150 // indirect
	golang.org/x/tools v0.1.10 // indirect
	golang.org/x/xerrors v0.0.0-20220411194840-2f41105eb62f // indirect
	lukechampine.com/uint128 v1.2.0 // indirect
	modernc.org/cc/v3 v3.35.26 // indirect
	modernc.org/ccgo/v3 v3.16.2 // indirect
	modernc.org/libc v1.15.1 // indirect
	modernc.org/mathutil v1.4.1 // indirect
	modernc.org/memory v1.0.7 // indirect
	modernc.org/opt v0.1.3 // indirect
	modernc.org/sqlite v1.16.0 // indirect
	modernc.org/strutil v1.1.1 // indirect
	modernc.org/token v1.0.0 // indirect
)
```
</details>

**Comment by vmihailenco** [comment]
Try to downgrade to `modernc.org/sqlite v1.15.0`. I am not sure what is changed in sqlite and how to fix it.

**Comment by alunegov** [comment]
Fixed in 23605846c20684e39bf1eaac50a2147a1b68a729
